src/main/java/slimeknights/tconstruct/gadgets/item/SlimeSlingItem.java

```diff
@@ -59,22 +68,92 @@ public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity en
       f = 6f;
     }
 
-    // check if player was targeting a block
-    RayTraceResult mop = rayTrace(worldIn, player, RayTraceContext.FluidMode.NONE);
-    if (mop.getType() == RayTraceResult.Type.BLOCK) {
-      // we fling the inverted player look vector
-      Vector3d vec = player.getLookVec().normalize();
-      player.addVelocity(vec.x * -f,
-        vec.y * -f / 3f,
-        vec.z * -f);
-
-      if (player instanceof ServerPlayerEntity) {
-        ServerPlayerEntity playerMP = (ServerPlayerEntity) player;
-        TinkerNetwork.getInstance().sendTo(new EntityMovementChangePacket(player), playerMP);
-      }
-
-      player.playSound(Sounds.SLIME_SLING.getSound(), 1f, 1f);
-      SlimeBounceHandler.addBounceHandler(player);
+    switch (this.slimeType) {
```

Comment:
This giant switch statement seems pretty messy. At this point I think I'd rather you just make separate item classes for each that extend a base as there is relatively little duplicated code in this method.

For the record, `EnumObject` has a builder syntax you can use in place of the `registerEnum` loop.